platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildContentGenerationContext.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+public class BuildContentGenerationContext {
```

Comment:
`BuildContentGenerationContext` is only introduced to avoid being too specific in the involved interfaces.

The `VersionCatalogDependencyRegistry` is just the only thing right now that keeps state across all projects generated in one invocation.

The `InitSettings` are created too early and transferred across the worker api so I did not add the `VersionCatalogDependencyRegistry` there